File: jvm-tools/XmlFormatter/src/main/kotlin/dev/vighnesh153/xml/formatter/fmt/FormatCommentNode.kt
package dev.vighnesh153.xml.formatter.fmt

import dev.vighnesh153.xml.formatter.ast.XmlCommentNode
import dev.vighnesh153.xml.formatter.ast.buildIndentation

internal fun XmlCommentNode.format(
    indentationLevel: Int,
    indentation: Int,
): String = "${
    buildIndentation(
        indentationLevel = indentationLevel,
        indentation = indentation
    )
}<!-- ${comment.tokenLiteral.trim()} -->"

File: nodejs-tools/nodejs-compiler-tools/formatter-xml/src/format_comment_node.ts
import { XmlCommentNode } from '@vighnesh153/parser-xml';
import { buildIndentationSpace } from './build_indentation_space.ts';

type FormatCommentNodeOptions = {
  commentNode: XmlCommentNode;
  indentationLevel: number;
  indentation: number;
};

export function formatCommentNode({ commentNode, indentationLevel, indentation }: FormatCommentNodeOptions): string {
  return (
    buildIndentationSpace({ indentationLevel, indentation }) + `<!-- ${commentNode.comment.tokenLiteral.trim()} -->`
  );
}
